ecs-init/cache/cache.go

```diff
@@ -57,6 +59,17 @@ func NewDownloader() *Downloader {
 		return downloader
 	}
 	downloader.metadata = ec2metadata.New(sessionInstance)
+
+	log.Debugf("Setting region for s3 client to: %s", downloader.getBucketRegion())
+	s3sessionInstance, err := session.NewSession(&aws.Config{
+		Credentials: credentials.AnonymousCredentials,
+		Region: aws.String(downloader.getBucketRegion()),
+	})
+	if err != nil {
```

Comment:
This will lead to silently consuming the error, which is less than ideal. Can you modify the signature of `NewDownloader` method return `error` and `Downloader` types and return the error when not `nil`?